internal/app/networkd/pkg/networkd/netconf.go

```diff
@@ -43,11 +44,31 @@ func buildOptions(device machine.Device, hostname string) (name string, opts []n
 		}
 
 		opts = append(opts, nic.WithAddressing(s))
+	case device.DHCP == false:
```

Comment:
I feel we can simplify this switch by structuring it as:

```
switch{
case CIDR != "":
   // assign static address
case DHCP:
   // use dhcp
default:
  // use dhcp is no vlans, otherwise keep without an address
}
```

Reply:
Agree, refactored to this approach